simulation/model.py

```diff
@@ -84,6 +90,34 @@ def increment_timestep(self):
             self.boiler_upgrade_scheme_spend_gbp
         )
 
+    def get_heat_pump_price_discount_schedule(
+        self,
+        heat_pump_price_discount_schedule: Optional[
+            List[Tuple[datetime.datetime, float]]
+        ],
+    ):
+
+        if heat_pump_price_discount_schedule:
+
+            discount_schedule = {
+                schedule[0]: schedule[1]
+                for schedule in heat_pump_price_discount_schedule
+            }
```

Comment:
A small suggestion - it's more readable to unpack the tuple instead of using indexing, e.g.

```
discount_schedule = {date: discount for date, discount in heat_pump_price_discount_schedule}
```